File: concepts/SOLID/PHP/5-DIP/c/src/main/payEmployeesMain.php
<?php

use app\notification\EmailSender;
use app\payment\PaymentProcessor;
use app\persistence\EmployeeFileSerializer;
use app\persistence\EmployeeRepository;

require_once '../../vendor/autoload.php';


$paymentProcessor = new PaymentProcessor(new EmployeeRepository(new EmployeeFileSerializer()), new EmailSender());
$totalPayments = $paymentProcessor->sendPayments();
 echo "Total payments " .  $totalPayments;

File: software-engineering-concepts/SOLID/Java/5-DIP/after/GlobomanticsHR/src/main/java/hr/main/PayEmployeesMain.java
package hr.main;

import hr.notifications.EmailSender;
import hr.payment.PaymentProcessor;
import hr.persistence.EmployeeFileRepository;
import hr.persistence.EmployeeFileSerializer;

public class PayEmployeesMain {

    /*
    Will take a couple of seconds to execute due to the
    sending of mails.
     */

    public static void main(String[] args) {
        PaymentProcessor paymentProcessor = new PaymentProcessor(
                new EmployeeFileRepository(new EmployeeFileSerializer()),
                        new EmailSender());
        int totalPayments = paymentProcessor.sendPayments();
        System.out.println("Total payments " + totalPayments);
    }
}
